### PR TITLE
Extract journey logic into a presenter

### PR DESCRIPTION
**Pull request by stevenwilkin**
A lot of this feels wrong but it takes the logic out of the view. What do you think chaps?

### CODE REVIEW

File: app/decorators/journey_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+class JourneyDecorator
+  attr_accessor :guide, :index, :current, :journey_guides
+
+  delegate :id, :label, :url, to: :guide
+
+  def self.decorate(journey_guides, current_guide)
+    i = 0
+    journey_guides.map do |guide|
```
Comment (andrewgarner) [inline app/decorators/journey_decorator.rb]: I believe you can do the following:

```rb
journey_guides.map.with_index do |guide, index|
  new(guide,
    journey_guides: journey_guides,
    current: (guide.id == current_guide.id),
    index: index)
end
```

**Comment by andrewgarner** [comment]
It's a good start!

File: app/decorators/journey_decorator.rb (hunk 1 above)
Comment (stevenwilkin) [inline app/decorators/journey_decorator.rb]: Every day's a school day, nice!

**Comment by andrewgarner** [comment]
Closing as this has been superseded by #210

**Pull request closed without merging**